cmd/agent/app/snmpwalk.go

```diff
@@ -0,0 +1,316 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gosnmp/gosnmp"
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultVersion = ""
+	defaultOID     = ""
+	defaultPort    = 161
+
+	// snmp v1 & v2c
+	defaultCommunityString = "public"
+
+	// snmp v3
+	defaultUserName      = ""
+	defaultAuthProtocol  = ""
+	defaultAuthKey       = ""
+	defaultPrivProtocol  = ""
+	defaultPrivKey       = ""
+	defaultContext       = "public"
+	defaultSecurityLevel = ""
+
+	// general communication options
+	defaultTimeout = 10 // Timeout better suited to walking
+	defaultRetries = 3
+)
+
+var (
+	// general
+	snmpVersion string
+	address     string
+	deviceIP    string
+	oid         string
+	port        uint16
+	value       uint64
+	setVersion  gosnmp.SnmpVersion
+	snmp        gosnmp.GoSNMP
+
+	// v1 & v2c
+	communityString string
+
+	// v3
+	user          string
+	authProtocol  gosnmp.SnmpV3AuthProtocol
+	authProt      string
+	authKey       string
+	privProtocol  gosnmp.SnmpV3PrivProtocol
+	privProt      string
+	privKey       string
+	snmpContext   string
+	msgFlags      gosnmp.SnmpV3MsgFlags
+	securityLevel string
+
+	// communication
+	retries int
+	timeout int
+)
+
+var (
+	snmpCmd = &cobra.Command{
+		Use:   "snmp",
+		Short: "Snmp tools",
+		Long:  ``,
+	}
+)
+
+func init() {
+
+	snmpWalkCmd.Flags().StringVarP(&snmpVersion, "snmp-version", "v", defaultVersion, "Specify SNMP version to use")
+
+	// snmp v1 or v2c specific
+	snmpWalkCmd.Flags().StringVarP(&communityString, "community-string", "C", "", "Set the community string")
+
+	// snmp v3 specific
+	snmpWalkCmd.Flags().StringVarP(&authProt, "auth-protocol", "a", defaultAuthProtocol, "Set authentication protocol (MD5|SHA|SHA-224|SHA-256|SHA-384|SHA-512)")
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏The suggested auth protocols seems to not match those defined later:


```
		if setVersion == gosnmp.Version3 {
			// Authentication Protocol
			switch strings.ToLower(authProt) {
			case "":
				authProtocol = gosnmp.NoAuth
			case "md5":
				authProtocol = gosnmp.MD5
			case "sha":
				authProtocol = gosnmp.SHA
			case "sha224":
				authProtocol = gosnmp.SHA224
			case "sha256":
				authProtocol = gosnmp.SHA256
			case "sha384":
				authProtocol = gosnmp.SHA384
			case "sha512":
				authProtocol = gosnmp.SHA512
			default:
				fmt.Printf("Unsupported authentication protocol: %s \n", authProt)
				cmd.Help() //nolint:errcheck
				os.Exit(1)
				return nil
			}
```

Reply:
The description messages are the same as those for the `net-snmp` `snmpwalk` which uses hyphens in the suggestions.

Reply:
Agree with @AlexandreYang on this, help will print unsupported syntax. We can 
1/ support both with and without hyphens format
2/ print supported values